src/neo4j_genai/exceptions.py

```diff
@@ -61,6 +61,12 @@ class Neo4jIndexError(Neo4jGenAiError):
     pass
 
 
+class Neo4jInsertionError(Neo4jGenAiError):
```

Comment:
I think we can also add this Exception to the docs

Reply:
I've updated `docs/source/api.rst` with it. Is there anything else I need to do to update the docs?

Reply:
Nope this looks great!